File: app/csharp/AccountingSystem.Domain/Models/Account.cs
using System.ComponentModel.DataAnnotations;

namespace AccountingSystem.Domain.Models
{
    /// <summary>
    /// 勘定科目マスタ
    /// </summary>
    public class Account
    {
        [Required(ErrorMessage = "勘定科目コードは必須です")]
        [StringLength(10, ErrorMessage = "勘定科目コードは10文字以内で入力してください")]
        public string AccountCode { get; set; } = string.Empty;

        [Required(ErrorMessage = "勘定科目名は必須です")]
        [StringLength(40, ErrorMessage = "勘定科目名は40文字以内で入力してください")]
        public string AccountName { get; set; } = string.Empty;

        [Required(ErrorMessage = "科目区分は必須です")]
        [StringLength(10, ErrorMessage = "科目区分は10文字以内で入力してください")]
        public string AccountType { get; set; } = string.Empty;

        public bool SumAccount { get; set; }
    }
}

File: app/fsharp/AccountingSystem.Domain/Models/Account.fs
namespace AccountingSystem.Domain.Models

/// <summary>
/// 勘定科目エンティティ
/// データベースの日本語カラム名と英語プロパティ名をマッピング
/// </summary>
[<CLIMutable>]
type Account = {
    AccountId: int option              // 勘定科目ID
    AccountCode: string                // 勘定科目コード
    AccountName: string                // 勘定科目名
    AccountNameKana: string option     // 勘定科目カナ
    AccountType: string                // 勘定科目種別
    IsSummaryAccount: bool             // 合計科目
    BsplType: string option            // BSPL区分
    TransactionElementType: string option  // 取引要素区分
    ExpenseType: string option         // 費用区分
    DisplayOrder: int                  // 表示順序
    IsAggregationTarget: bool          // 集計対象
    TaxCode: string option             // 課税取引コード
    Balance: decimal                   // 残高
}

/// Account エンティティのファクトリ関数
module Account =
    let create accountCode accountName accountType isSummaryAccount =
        {
            AccountId = None
            AccountCode = accountCode
            AccountName = accountName
            AccountNameKana = None
            AccountType = accountType
            IsSummaryAccount = isSummaryAccount
            BsplType = None
            TransactionElementType = None
            ExpenseType = None
            DisplayOrder = 0
            IsAggregationTarget = true
            TaxCode = None
            Balance = 0m
        }
